src/new_storage/error.rs

```diff
@@ -29,3 +33,35 @@ pub enum StorageError {
     #[error("{0}")]
     Internal(#[from] anyhow::Error),
 }
+
+impl From<StorageError> for DatenLordError {
+    fn from(value: StorageError) -> Self {
+        match value {
+            StorageError::OutOfRange { maximum, found } => DatenLordError::InternalErr {
```

Comment:
I think most places use StorageError::OutOfRange can be replace with assert, thus removing one error type